drivers/SmartThings/matter-switch/src/init.lua

```diff
@@ -1125,26 +1185,57 @@ local function occupancy_attr_handler(driver, device, ib, response)
   device:emit_event(ib.data.value == 0x01 and capabilities.motionSensor.motion.active() or capabilities.motionSensor.motion.inactive())
 end
 
+local function available_endpoints_handler(driver, device, ib, response)
+  local electrical_sensor_eps = device:get_field(ELECTRICAL_SENSOR_EPS)
+  local avail_eps_req_complete = true
+  local electrical_tags_per_ep = {}
+  table.sort(ib.data.elements)
+  for _, info in ipairs(electrical_sensor_eps) do
+    if info.endpoint_id == ib.endpoint_id then
+      info.availableEndpoints = ib.data.elements
+    elseif info.availableEndpoints == false then -- an endpoint is found that hasn't been updated through this handler.
+      avail_eps_req_complete = false
+    end
+
+    if avail_eps_req_complete then
+      local electrical_tags = ""
+      if info.power then electrical_tags = electrical_tags .. "-power" end
+      if info.energy then electrical_tags = electrical_tags .. "-energy-powerConsumption" end
+      electrical_tags_per_ep[info.availableEndpoints[1].value] = electrical_tags -- set tags on first available endpoint
+    end
+  end
```

Comment:
```suggestion
  if electrical_sensor_eps then
    for _, info in ipairs(electrical_sensor_eps) do
      if info.endpoint_id == ib.endpoint_id then
        info.availableEndpoints = ib.data.elements
      elseif info.availableEndpoints == false then -- an endpoint is found that hasn't been updated through this handler.
        avail_eps_req_complete = false
      end

      if avail_eps_req_complete then
        local electrical_tags = ""
        if info.power then electrical_tags = electrical_tags .. "-power" end
        if info.energy then electrical_tags = electrical_tags .. "-energy-powerConsumption" end
        electrical_tags_per_ep[info.availableEndpoints[1].value] = electrical_tags -- set tags on first available endpoint
      end
    end
  end
```
`ipairs` would throw an error if this is nil

Reply:
FWIW it never will be if this is run, still would be safer